cpu_scheduling_algorithms/non_preemptive_sjf_scheduling.cpp

```diff
@@ -0,0 +1,316 @@
+ * detailed description on SJF scheduling </a>
+ * <a href="https://github.com/LakshmiSrikumar">Author : Lakshmi Srikumar </a>
+ */
+
+#include <algorithm>      /// for sorting
+#include <cassert>        /// for assert
+#include <iomanip>        /// for formatting the output
+#include <iostream>       /// for IO operations
+#include <queue>          /// for std::priority_queue
+#include <random>         /// random number generation
+#include <unordered_set>  /// for std::unordered_set
+#include <vector>         /// for std::vector
+
+using std::cin;
+using std::cout;
+using std::endl;
+using std::get;
+using std::left;
+using std::make_tuple;
+using std::priority_queue;
+using std::tuple;
+using std::unordered_set;
+using std::vector;
+
+/**
+ * @brief Comparator function for sorting a vector
+ * @tparam S Data type of Process ID
+ * @tparam T Data type of Arrival time
+ * @tparam E Data type of Burst time
+ * @param t1 First tuple<S,T,E>t1
+ * @param t2 Second tuple<S,T,E>t2
+ * @returns true if t1 and t2 are in the CORRECT order
+ * @returns false if t1 and t2 are in the INCORRECT order
+ */
+template <typename S, typename T, typename E>
+bool sortcol(tuple<S, T, E>& t1, tuple<S, T, E>& t2) {
+    if (get<1>(t1) < get<1>(t2) ||
+        (get<1>(t1) == get<1>(t2) && get<0>(t1) < get<0>(t2))) {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * @class Compare
+ * @brief Comparator class for priority queue
+ * @tparam S Data type of Process ID
+ * @tparam T Data type of Arrival time
+ * @tparam E Data type of Burst time
+ */
+template <typename S, typename T, typename E>
+class Compare {
+ public:
+    /**
+     * @param t1 First tuple
+     * @param t2 Second tuple
+     * @brief A comparator function that checks whether to swap the two tuples
+     * or not.
+     * <a
+     * href="https://www.geeksforgeeks.org/comparator-class-in-c-with-examples/">
+     * detailed description of comparator </a>
+     * @returns true if the tuples SHOULD be swapped
+     * @returns false if the tuples SHOULDN'T be swapped
+     */
+    bool operator()(tuple<S, T, E, double, double, double>& t1,
+                    tuple<S, T, E, double, double, double>& t2) {
+        // Compare burst times for SJF
+        if (get<2>(t2) < get<2>(t1)) {
+            return true;
+        }
+        // If burst times are the same, compare arrival times
+        else if (get<2>(t2) == get<2>(t1)) {
+            return get<1>(t2) < get<1>(t1);
+        }
+        return false;
+    }
+};
+
+/**
+ * @class SJF
+ * @brief Class which implements the SJF scheduling algorithm
+ * @tparam S Data type of Process ID
+ * @tparam T Data type of Arrival time
+ * @tparam E Data type of Burst time
+ */
+template <typename S, typename T, typename E>
+class SJF {
+    /**
+     * Priority queue of schedules(stored as tuples) of processes.
+     * In each tuple
+     * @tparam 1st element: Process ID
+     * @tparam 2nd element: Arrival Time
```

Comment:
why are we re documenting template parameters here. One definiton is enough

Reply:
In the earlier conversation , you had said to document what the tuples contain .

Since there are no more changes to be done ., could you please accept this repo and merge it